test/test_mock.py

```diff
@@ -283,6 +283,23 @@ def test_called_CoroutineMock_returns_MagicMock(self):
         mock = asynctest.mock.CoroutineMock()
         self.assertIsInstance(run_coroutine(mock()), asynctest.mock.MagicMock)
 
+    def test_awaited_CoroutineMock_sets_awaited(self):
```

Comment:
Can you add more tests:
* when result is awaitable?
* when awaiting the result raises an exception?